src/controllers/account/index.ts

```diff
@@ -1,45 +1,110 @@
-    // const resetToken = await AccountUtilities.splitSelectorVerifier(token)
-    // DB lookup first 16 bytes provided (selector)
-    // if found, hash next 16 bytes provided (verifier)
-    // const verifierHash = await AccountUtilities.getVerifierHash(resetToken)
-    // compare hash with verifier hash in DB
+    ctx.assert(email, 400, Strings.EmailIsRequired)
+    
+    // Lookup email in mysql db to make sure it's a registered user
+    const existingAccount = await getAccount({ email })
+    // If an existing account doesnt exist, the user was never registered
+    if (!existingAccount) {
+      return ctx.body = { ok: false, reason: 'no user exists' }
+    }
+
+    // Generate token, store it
+    const emailToken = await generateAndStoreToken(ctx, false, true, {})
     try {
-      // const storedVerifierHash = await AccountUtilities.getStoredVerifierHash(resetToken)
-      // const match = await AccountUtilities.compareHashes(storedVerifierHash, verifierHash)
-      // if (!match) {
-      //   // if hash does not match, 401 and delete token
-      //   await AccountUtilities.deleteSelector(resetToken)
-      //   ctx.assert(match, 401)
-      // }
+      await Methods.sendMagicLinkEmail(email, emailToken)
+      ctx.body = { ok: true }
     } catch (e) {
-      ctx.throw(401)
+      console.log(e)
+      ctx.body = { ok: false, message: e.message || 'unknown error' }
     }
+    return
+  }
+
+  // Endpoint to create hash to send in email to the user on register
+  public static async register (ctx: BaseContext) {
+    const { email, role } = ctx.request.body
+    ctx.assert(email, 400, Strings.EmailIsRequired)
+    ctx.assert(role, 400, Strings.RoleIsRequired)
 
-    // if match and hasn't expired, reset the password
-    return AccountController.resetPassword(ctx)
+    const sessionMeta = { 
+      firstName: ctx.request.body.firstName, 
+      lastName: ctx.request.body.lastName 
+    }
+    // Generate token, store it
+    const emailToken = await generateAndStoreToken(ctx, false, false, sessionMeta)
+    try {
+      await Methods.sendMagicLinkEmail(email, emailToken)
+      ctx.body = { ok: true }
+    } catch (e) {
+      ctx.body = { ok: false, message: e.message || 'unkonwn error' }
+    }
   }
-  public static async resetPassword (ctx: BaseContext) {
-    ctx.status = 200
-    ctx.body = { ok: true }
+
+  // Endpoint to verify hash from email magic link and send back auth token
+  public static async verifyAuth (ctx: BaseContext) {
+    const { email, auth } = ctx.request.body
+    ctx.assert(email, 400, Strings.EmailIsRequired)
+    ctx.assert(auth, 400, Strings.AuthIsRequired)
+
+    // Look up hash sent in request in the db 
+    try {
+      const authToken = await splitSelectorVerifier(ctx.request.body.auth)
+      if (!authToken) return ctx.status = 401
+
+      const sessionInfo = await getStoredSession(authToken)
+      // delete the short term session
+      await deleteSelector(sessionInfo)
```

Comment:
Can probably remove `await` here because we don't need to block the thread while the selector is deleted from Dynamo.

Reply:
done